deepchem/molnet/load_function/delaney_datasets.py

```diff
@@ -34,95 +38,102 @@ def load_delaney(featurizer='ECFP',
   - "measured log solubility in mols per litre" - Log-scale water solubility
     of the compound, used as label
 
+  Parameters
+  ----------
+  featurizer: Featurizer or str
+    the featurizer to use for processing the data.  Alternatively you can pass
+    one of the names from dc.molnet.defaults.featurizers as a shortcut.
+  splitter: Splitter or str
+    the splitter to use for splitting the data into training, validation, and
+    test sets.  Alternatively you can pass one of the names from
+    dc.molnet.defaults.splitters as a shortcut.  If this is None, all the data
+    will be included in a single dataset.
+  reload: bool
+    if True, the first call for a particular featurizer and splitter will cache
+    the datasets to disk, and subsequent calls will reload the cached datasets.
+  move_mean: bool
+    if True, all the data is shifted so the training set has a mean of zero.
+  data_dir: str
+    a directory to save the raw data in
+  save_dir: str
+    a directory to save the dataset in
+
   References
   ----------
   .. [1] Delaney, John S. "ESOL: estimating aqueous solubility directly from
      molecular structure." Journal of chemical information and computer
      sciences 44.3 (2004): 1000-1005.
   """
-  # Featurize Delaney dataset
-  logger.info("About to featurize Delaney dataset.")
+  if 'split' in kwargs:
+    splitter = kwargs['split']
+    logger.warning("'split' is deprecated.  Use 'splitter' instead.")
+  if isinstance(featurizer, str):
+    featurizer = dc.molnet.defaults.featurizers[featurizer]
+  if isinstance(splitter, str):
+    splitter = dc.molnet.defaults.splitters[splitter]
   if data_dir is None:
     data_dir = DEFAULT_DIR
   if save_dir is None:
     save_dir = DEFAULT_DIR
+  tasks = ['measured log solubility in mols per litre']
+
+  # Try to reload cached datasets.
+
   if reload:
-    save_folder = os.path.join(save_dir, "delaney-featurized")
+    featurizer_name = str(featurizer.__class__.__name__)
```

Comment:
One thought here is perhaps we could stringify it more intelligently so we can store multiple versions that use the same featurizer but different parameters. Looking at the string representation:

```
In [5]: feat = dc.feat.CircularFingerprint(size=1024)                  

In [6]: feat                       
Out[6]: CircularFingerprint[radius=2, size=1024, chiral=False, bonds=True, features=False, sparse=False, smiles=False]
```
Perhaps this could be
```
featurizer_name = CircularFingerprint_radius_2_size_1024_chiral_False_bonds_True_features_False_sparse_False_smiles_False
```
And similarly for splitters

Reply:
Done.  I copied over the implementations of `__str__()` and `__repr()__` from Featurizer to Splitter.